### ISSUE TITLE
discoverd: Stuck "Waiting for store to catchup" during deployment

### ISSUE THREAD

**Issue by lmars** [flynn/flynn#2698]
Discoverd deployments are currently broken in CI, with the entries like the following in the discoverd logs (from [this build failure](https://ci.flynn.io/builds/20160413005440-27752ff7)):

```
2016/04/13 01:06:00 checking for existing discoverd process at http://10.69.25.7:1111
2016/04/13 01:06:00 discoverd responding at http://10.69.25.7:1111 taking over
2016/04/13 01:06:00 Created deployment
t=2016-04-13T01:06:00+0000 lvl=info msg="sent state change" app=containerinit fn=StreamState state=running err= exitStatus=-1
2016/04/13 01:06:04 marked 10.69.25.7:1111 as performing in deployent
2016/04/13 01:06:04 starting proxy DNS server
2016/04/13 01:06:04 discoverd listening for DNS on 100.100.9.1:53
2016/04/13 01:06:06 [INFO] raft: Node at 10.69.25.7:1111 [Follower] entering Follower state
2016/04/13 01:06:06 Waiting for store to catchup, current: 0 target: 140
2016/04/13 01:06:06 Waiting for store to catchup, current: 0 target: 140
2016/04/13 01:06:07 Waiting for store to catchup, current: 0 target: 140
2016/04/13 01:06:07 Waiting for store to catchup, current: 0 target: 140
2016/04/13 01:06:07 Waiting for store to catchup, current: 0 target: 140
2016/04/13 01:06:07 Waiting for store to catchup, current: 0 target: 140
2016/04/13 01:06:07 Waiting for store to catchup, current: 0 target: 140
2016/04/13 01:06:07 Waiting for store to catchup, current: 0 target: 140
2016/04/13 01:06:07 Waiting for store to catchup, current: 0 target: 140
2016/04/13 01:06:07 Waiting for store to catchup, current: 0 target: 140
2016/04/13 01:06:07 Waiting for store to catchup, current: 0 target: 140
2016/04/13 01:06:07 Waiting for store to catchup, current: 0 target: 140
2016/04/13 01:06:08 Waiting for store to catchup, current: 0 target: 140
2016/04/13 01:06:08 Waiting for store to catchup, current: 0 target: 140
2016/04/13 01:06:08 Waiting for store to catchup, current: 0 target: 140
2016/04/13 01:06:08 Waiting for store to catchup, current: 0 target: 140
2016/04/13 01:06:08 Waiting for store to catchup, current: 0 target: 140
2016/04/13 01:06:08 Waiting for store to catchup, current: 0 target: 140
2016/04/13 01:06:08 [WARN] raft: Heartbeat timeout reached, starting election
2016/04/13 01:06:08 [INFO] raft: Node at 10.69.25.7:1111 [Candidate] entering Candidate state
2016/04/13 01:06:08 [DEBUG] raft: Votes needed: 2
2016/04/13 01:06:08 [DEBUG] raft: Vote granted from 10.69.25.7:1111. Tally: 1
2016/04/13 01:06:08 [DEBUG] raft: Vote granted from 10.69.25.8:1111. Tally: 2
2016/04/13 01:06:08 [INFO] raft: Election won. Tally: 2
2016/04/13 01:06:08 [INFO] raft: Node at 10.69.25.7:1111 [Leader] entering Leader state
2016/04/13 01:06:08 [INFO] raft: pipelining replication to peer 10.69.25.8:1111
2016/04/13 01:06:08 [DEBUG] raft: Node 10.69.25.7:1111 updated peer set (2): [10.69.25.7:1111 10.69.25.6:1111 10.69.25.8:1111]
2016/04/13 01:06:08 [INFO] raft: pipelining replication to peer 10.69.25.6:1111
2016/04/13 01:06:08 Waiting for store to catchup, current: 1 target: 140
[discoverd] 2016/04/13 01:06:08 enforce expiry: discoverd: new leader, waiting for 2x TTL
2016/04/13 01:06:08 Waiting for store to catchup, current: 1 target: 140
2016/04/13 01:06:08 Waiting for store to catchup, current: 1 target: 140
2016/04/13 01:06:09 Waiting for store to catchup, current: 1 target: 140
...
2016/04/13 01:24:19 Waiting for store to catchup, current: 1 target: 140
2016/04/13 01:24:19 Waiting for store to catchup, current: 1 target: 140
2016/04/13 01:24:19 Waiting for store to catchup, current: 1 target: 140
2016/04/13 01:24:19 Waiting for store to catchup, current: 1 target: 140
2016/04/13 01:24:19 Waiting for store to catchup, current: 1 target: 140
```

In this instance, the store remained at `current: 1 target: 140` for 18 mins (until the CI build stopped).

@josephglanville any thoughts?

**Comment by titanous** [comment]
We should check if this related to the update of the Raft library.

**Comment by josephglanville** [comment]
Yep this does appear to be caused by update to the Raft library. I think it's due to the changes to close the transport on Raft shutdown possibly. I should have a fix sometime today.

**Comment by josephglanville** [comment]
This should also have been fixed by #2724
